docs/docs/custom-fields.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Custom fields
+tags:
+  - content-type-builder
+  - plugins
+---
+
+# Custom fields
+
+## Summary
+
+Custom fields provide a way to replace the inputs of existing Strapi types to improve the content editing experience.
+
+## Detailed design
+
+A custom field needs to be registered in both the admin and server.
+
+### Server
+
+We need to ensure that an attribute using a custom field is valid, we expose a new `customFields` object with a `register` method on the `Strapi` instance.
+
+Here's how you register a Custom Field on the server:
+
+```ts
+interface CustomFieldServerOptions {
+  // The name of the custom field
+  name: string;
+  // The name of the plugin creating the custom field
+  plugin?: string;
+  // The existing Strapi data type the custom field will use
+  type: string;
+}
+
+strapi.customFields.register(
+  options: CustomFieldServerOptions | CustomFieldServerOptions[]
+);
+```
+
+The custom field will be added to Strapi during the server [register lifecycle](https://docs.strapi.io/developer-docs/latest/developer-resources/plugin-api-reference/server.html#register).
+
+### Admin
+
+To register a custom field on the server, see [documentation](https://docs.strapi.io/developer-docs/latest/development/custom-fields.html#registering-a-custom-field-on-the-server).
```

Comment:
I think this should go above under the Server sub heading and then I think you can get rid of the code block starting on line 24 since the docs link should explain all that